src/lib/state/transactions.ts

```diff
@@ -21,20 +21,18 @@ export interface ApprovalTransactionInfo extends BaseTransactionInfo {
 export interface SwapTransactionInfo extends BaseTransactionInfo {
   type: TransactionType.SWAP
   tradeType: TradeType
-  inputCurrencyAddress: string
-  outputCurrencyAddress: string
+  inputCurrencyAmount: CurrencyAmount<Currency>
+  outputCurrencyAmount: CurrencyAmount<Currency>
 }
```

Comment:
confirming we don't need addresses here?

Reply:
yep the currency amount can give us token which can give us address